src/main/java/ArrayIndexedBag.java

```diff
@@ -0,0 +1,174 @@
+     *
+     * @param start Index of where shifting starts/stops and where
+     *              space is opened.
+     */
+    private void shiftRight(int start) {
+        for (int i = rear; i > start; --i) {
+            bag[i] = bag[i - 1];
+        }
+        bag[start] = null;
+    }
+
+    @Override
+    public void add(T element) {
+        add(rear, element);
+    }
+
+    @Override
+    public void add(int index, T element) {
+        if (index > rear) {
+            throw new IndexOutOfBoundsException(String.format("Bag has no index %d to add to.", index));
+        }
+        if (size() == bag.length) {
+            expandCapacity();
+        }
+        shiftRight(index);
+        bag[index] = element;
+        rear++;
+    }
+
+    @Override
+    public void set(int index, T element) {
+        if (index >= rear) {
+            throw new IndexOutOfBoundsException(String.format("Bag has no element at index %d.", index));
+        }
+        bag[index] = element;
+    }
+
+    @Override
+    public T get(int index) {
+        if (index >= rear) {
+            throw new IndexOutOfBoundsException(String.format("Bag has no element at index %d.", index));
+        }
+        return bag[index];
+    }
+
+    @Override
+    public T remove(int index) {
+        if (index >= rear) {
+            throw new IndexOutOfBoundsException(String.format("Bag has no element at index %d.", index));
+        }
+        T returnElement = bag[index];
+        shiftLeft(index);
+        rear--;
+        return returnElement;
+    }
+
+    @Override
+    public int indexOf(T target) {
+        for (int i = 0; i < bag.length; ++i) {
+            if (bag[i].equals(target)) {
+                return i;
+            }
+        }
+        return NOT_FOUND;
+    }
+
+    @Override
+    public T remove(T element) {
+        if (isEmpty()) {
+            throw new NoSuchElementException("Removing from an empty bag.");
+        }
+        int removeIndex = indexOf(element);
+        if (removeIndex == NOT_FOUND) {
+            throw new NoSuchElementException("Element not contained in bag.");
+        }
+        return remove(removeIndex);
+    }
+
+    @Override
+    public boolean contains(T target) {
+        return indexOf(target) != NOT_FOUND;
+    }
+
+    @Override
+    public int getCount(T target) {
+        int count = 0;
+        Iterator<T> it = this.iterator();
+        while (it.hasNext()) {
+            if (it.next().equals(target)) {
+                count++;
+            }
+        }
+        return count;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return rear == 0;
+    }
+
+    @Override
+    public int size() {
+        return rear;
+    }
+
+    @Override
+    public Iterator<T> iterator() {
+        return new ArrayIterator<>(bag, rear);
+    }
+
+    public String toString() {
+        Iterator<T> it = this.iterator();
+        StringBuilder builder = new StringBuilder();
+        builder.append("Front --> ");
```

Comment:
Remove? Even though there's a "front" for an indexed bag, it seems a little out of place. This method is still `Bag#toString`.

Reply:
There is currently no implementation of a bag.

Reply:
I fail to see how that is related?

Reply:
Are you referring to `builder.append("Front --> ");` specifically? I originally thought you meant the whole `toString`, which threw me.

Reply:
Just the `Front` part.

Reply:
If I remove it from here, I will remove from `SortedBag` too since we should assume left -> right reading.

Reply:
Think it makes sense. There's no real front to a _bag_ -- this is `Bag#toString` after all. Inheritance should extend behaviours, not really change existing ones (at least, on the interface level).